core/test/utils/matrix_utils.hpp

```diff
@@ -47,108 +47,221 @@ namespace test {
 
 
 /**
```

Comment:
If you decide to move this out of test, maybe also change the namespace `test` to `utils`?
Also, it seems like you no longer need to include `core/test/utils/value_generator.hpp`.